### ISSUE TITLE
Datashare issue for users belongs to two or more teams

### ISSUE THREAD

**Issue by manjulaK** [awslabs/aws-dataall#669]
### Describe the bug

We have noticed that a user is  a part of 2 teams (eg producer , consumer) 2) producer team has a dataset-test. 3) you request permissions for **consumer team** for dataset-test 4) you won't see the submit button .

### How to Reproduce

```
*P.S. Please do not attach files as it's considered a security risk. Add code snippets directly in the message body as much as possible.*
```

### Expected behavior

When a user belongs to 2 teams and requests permissions for a dataset for a team that does not own the dataset they should be able to request access.

### OS

All

### Python version

3.1

### AWS data.all version

all

**Comment by dlpzx** [comment]
Hi @manjulaK, thanks for raising an issue. We will look into it, but you are more than welcome to start working on it. The implementation of a mechanism that adapts the sharing so that a user can be approver and requester at the same time is not a complicated implementation but our bandwidth is limited at the moment

**Comment by dlpzx** [comment]
Merged and released with v2.1.0 🚀
